internal/metric/wrapper.go

```diff
@@ -0,0 +1,208 @@
+package metric
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3"
+	"github.com/pkg/errors"
+	"google.golang.org/api/iterator"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+	duration "google.golang.org/protobuf/types/known/durationpb"
+	timestamp "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// Client represents a wrapper around the Cloud Monitoring package.
+type Client interface {
+	Latency(ctx context.Context, filter Filter, startTime time.Time) (float64, error)
+}
+
+// API is a wrapper for the Cloud Monitoring package.
+type API struct {
+	Client  *monitoring.MetricClient
+	Project string
+}
+
+// Filter is the filter for the query.
+type Filter struct {
+	service  string
+	revision string
+	query    string
+}
+
+// Metric types.
+const (
+	RequestLatencies = "run.googleapis.com/request_latencies"
+	RequestCount     = "run.googleapis.com/request_count"
+)
+
+type alignReduce int32
+
+// Series aligner and cross series reducer types (for latency).
+const (
+	Align99Reduce99 alignReduce = 1
```

Comment:
I'm not aware of aligning and reducing terminology. What's the difference?

Reply:
When obtaining metrics, you set an interval for the time you're interested in (in the Cloud Run Metrics graph, for instance, you have 1 day, 1 hour, etc.). Of course, this interval is customizable.

The interval can be broken up in pieces determined by `alignmentPeriod`. For instance, you can break an interval of 1h in pieces of 20 minutes each. Doing so you get a total of `interval / alignmentPeriod` points. Each point represents what happened during `alignmentPeriod` seconds. Of course, if you set `alignmentPeriod=1200s` (20 minutes), there might be a lot of requests during those 20 minutes. How do you get one single point for entire 20-minute period? By using the per series **aligner**. The aligner is basically the function that determines how to aggregate all those requests in the 20-minute period and pick one value.

But you get a total of `interval / alignmentPeriod` points. That's where the cross series reducer helps you get one single point based on a grouping fields like response code class (2xx, 5xx, etc.).